tsl/src/compression/api.c

```diff
@@ -710,6 +710,15 @@ recompress_chunk_impl(Chunk *chunk, Oid *uncompressed_chunk_id, bool recompress)
 	/* TODO: optimize cases where settings differ but recompression is still possible */
 	else if (recompress && ts_compression_settings_equal_with_defaults(ht_settings, chunk_settings))
 	{
+		if (!ts_guc_enable_in_memory_recompression)
+		{
+			ereport(NOTICE,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
```

Comment:
```suggestion
					(errcode(ERRCODE_WARNING),
```
Since this is a warning and not an error this is not the right code.